test/performance/MapViewIntegrationTest.ts

```diff
@@ -0,0 +1,131 @@
+import { OmvTileDecoder } from "@here/harp-omv-datasource/index-worker";
+import { getTestResourceUrl, getWebGLRendererStub } from "@here/harp-test-utils";
+import { FontCatalog } from "@here/harp-text-canvas";
+import { LoggerManager, PerformanceTimer } from "@here/harp-utils";
+import { AbortController } from "abort-controller";
+import { JSDOM } from "jsdom";
+import * as sinon from "sinon";
+import * as THREE from "three";
+
+// tslint:disable:no-empty
+
+declare const global: any;
+describe("MapView integration headless test", () => {
+    const inNodeContext = typeof window === "undefined";
+    let sandbox: sinon.SinonSandbox;
+    let clearColorStub: sinon.SinonStub;
+    // tslint:disable-next-line:no-unused-variable
+    let webGlStub: sinon.SinonStub;
+    // tslint:disable-next-line:no-unused-variable
+    let fontStub: sinon.SinonStub;
+    let addEventListenerSpy: sinon.SinonStub;
+    let removeEventListenerSpy: sinon.SinonStub;
+    let canvas: HTMLCanvasElement;
+    const CANVAS_WIDTH = 1000;
+    const CANVAS_HEIGHT = 1000;
+    const jsdom = new JSDOM();
+    beforeEach(() => {
+        sandbox = sinon.createSandbox();
+        clearColorStub = sandbox.stub();
+        webGlStub = sandbox
+            .stub(THREE, "WebGLRenderer")
+            .returns(getWebGLRendererStub(sandbox, clearColorStub));
+        const image = jsdom.window.document.createElement("img");
+        image.setAttribute("width", "1");
+        image.setAttribute("height", "1");
+        const emptyTexture = new THREE.Texture(image);
+        fontStub = sandbox.stub(FontCatalog, "loadTexture").returns(Promise.resolve(emptyTexture));
+        if (inNodeContext) {
+            const theGlobal: any = global;
+            theGlobal.window = {
+                window: { devicePixelRatio: 10 },
+                location: { href: "" }
+            };
+            theGlobal.navigator = {};
+            theGlobal.requestAnimationFrame = () => {};
+            theGlobal.performance = {
+                now: () => {
+                    return PerformanceTimer.now();
+                }
+            };
+            theGlobal.document = jsdom.window.document;
+            // Override the AbortController that is given by the web browser.
+            theGlobal.AbortController = AbortController;
+        }
+        addEventListenerSpy = sinon.stub();
+        removeEventListenerSpy = sinon.stub();
+        canvas = ({
+            clientWidth: CANVAS_WIDTH,
+            clientHeight: CANVAS_HEIGHT,
+            addEventListener: addEventListenerSpy,
+            removeEventListener: removeEventListenerSpy
+        } as unknown) as HTMLCanvasElement;
+    });
+    afterEach(() => {
+        sandbox.restore();
+        if (inNodeContext) {
+            delete global.window;
+            delete global.requestAnimationFrame;
+            delete global.cancelAnimationFrame;
+            delete global.navigator;
+            delete global.performance;
+        }
+    });
```

Comment:
add `mapView.dispose`, in `afterEach` so MapView instance releases all resources even if failed.

Reply:
the MapView instance is created in the it(...), so it should be released anyway once done is called? 

Or should i create it in the beforeEach ?